src/core/Rule.ml

```diff
@@ -358,6 +358,23 @@ and extract_reduction = Separate | Concat [@@deriving show]
 and extract_transform = NoTransform | Unquote | ConcatJsonArray
 [@@deriving show]
 
+(*****************************************************************************)
+(* Secrets mode *)
+(*****************************************************************************)
+
+(*TODO: actualy figure this out*)
+type secrets_spec = {
+  (* secrets: *)
+  secrets : formula list;
+  (* request: *)
+  url : string;
+  method_ : string;
+  headers : (string * string) list;
+  (* response: *)
```

Comment:
same, better to have intermediate record for response since I guess it will grow

Reply:
Add intermediate records for `response` & `request`